kratos/containers/model.cpp

```diff
@@ -27,109 +27,196 @@
 
 namespace Kratos
 {
-    void Model::AddModelPart( ModelPart::Pointer pModelPart) //TODO: DEPRECATED. to be removed. this is TEMPORARY
+    void Model::Reset()
+    {
+        mRootModelPartMap.clear();
+    }
+    
+    ModelPart& Model::CreateModelPart( const std::string ModelPartName ) 
     {
         KRATOS_TRY
-
-        //TODO: flat map should disappear in the future!!
-        auto search = mflat_map.find(pModelPart->Name());
-        if( search == mflat_map.end())
+        
+        std::cout << "within CreateModelPart address of Model is " <<  &(*this) << std::endl;
+        
+        auto search = mRootModelPartMap.find(ModelPartName);
+        if( search == mRootModelPartMap.end())
         {
-            mflat_map[pModelPart->Name()] = pModelPart.get();
-
-            //walk the submodelparts
-            for(auto& part : pModelPart->SubModelParts())
-                AddModelPartRawPointer(&part);
+            std::cout << ModelPartName << std::endl;
+            mRootModelPartMap[ModelPartName] = std::unique_ptr<ModelPart>(new ModelPart(ModelPartName));
```

Comment:
Kratos::make_unique

Reply:
i forked this branch prior to the definition of Kratos::make_unique. i cannot use it until i merge master into this, and i am not ready for that right now.

i'll do when (and if) i merge this to the master